docs/developer_guide/schemas.rst

```diff
@@ -0,0 +1,374 @@
+
+The ``.__init__()`` of each ``DataInterface`` has a unique call signature, requiring the source data to be input in a
+specific way. Some require a ``file_path``, some a ``directory_path``, and some require additional arguments that are
+necessary to read the source data properly. The same is true for the ``.run_conversion()`` method, where some classes
+have unique additional optional arguments. To communicate these requirements and options with an ``NWBConverter`` and
+with interfacing software, the expected form of the input is defined using the JSON Schema language.
+``DataInterface`` classes define the method
+:py:func:`~neuroconv.datainterfaces.basedatainterface.BaseDataInterface.get_source_schema()`, which is
+responsible for returning the source schema as a JSON schema dictionary. By default, these dictionaries are
+automatically derived from the call signatures of a function, and express similar information to a type-hinted call
+signature, but can also be used to define and validate inputs that are arbitrarily deep nested dictionaries. For
+example, :py:func:`SpikeGLXRecordingInterface.__init__()` has the call signature
+
+.. code-block:: python
+
+    def __init__(
+        self,
+        file_path: str,
+        stub_test: bool = False,
+        spikeextractors_backend: bool = False,
+        verbose: bool = True,
+    )
+
+Calling
+:py:func:`~neuroconv.datainterfaces.ecephys.spikeglx.spikeglxdatainterface.SpikeGLXRecordingInterface.get_source_schema()`
+gives the following output, which is derived from the call signature and expresses identical constraints.
+
+.. code-block:: json
+
+    {
+      "required": [
+        "file_path"
+      ],
+      "properties": {
+        "file_path": {
+          "format": "file",
+          "type": "string",
+          "description": "Path to SpikeGLX file."
+        },
+        "stub_test": {
+          "type": "boolean",
+          "default": false
+        },
+        "spikeextractors_backend": {
+          "type": "boolean",
+          "default": false
+        },
+        "verbose": {
+          "type": "boolean",
+          "default": true
+        }
+      },
+      "type": "object",
+      "additionalProperties": false
+    }
+
+
+``PhySortingInterface.get_source_schema()`` returns a similar schema:
+
+.. code-block:: json
+
+    {
+      "required": [
+        "folder_path"
+      ],
+      "properties": {
+        "folder_path": {
+          "format": "directory",
+          "type": "string"
+        },
+        "exclude_cluster_groups": {
+          "type": "array"
+        },
+        "verbose": {
+          "type": "boolean",
+          "default": true
+        },
+        "spikeextractors_backend": {
+          "type": "boolean",
+          "default": false
+        }
+      },
+      "type": "object",
+      "additionalProperties": false
+    }
+
+A ``ExampleNWBConverter`` that combines these ``DataInterface``s will combine the JSON Schema using the keys of the
```

Comment:
```suggestion
An ``ExampleNWBConverter`` that combines these ``DataInterface``s will combine the JSON Schema using the keys of the
```
Matter of opinion, but I imaging reading the Python class as if it were a part of the sentence, like 'An example NWBConverter...'